aten/src/TH/vector/AVX2.cpp

```diff
@@ -127,4 +127,30 @@ void THFloatVector_sigmoid_AVX2(float *y, const float *x, const ptrdiff_t n) {
   }
 }
 
+void THFloatVector_tanh_AVX2(float* y, const float* x, const ptrdiff_t n) {
+  ptrdiff_t i;
+  const __m256 one = _mm256_set1_ps(1.0f);
+  __m256 YMM0, YMM1, YMM2, YMM3;
```

Comment:
Is there a reason to pre-declare the vector registers?

Reply:
I was following what other functions in the `AVX2.cpp` was doing. We can declare them inside the loop as well.